app/models/git_repository.rb

```diff
@@ -155,6 +183,36 @@ def handling_worktree_errors
     raise MiqException::Error, err.message
   end
 
+  def git_lock_filename
+    @git_lock_filename ||= LOCKFILE_DIR.join(id.to_s)
+  end
+
+  def acquire_git_lock
+    return false if defined?(@git_lock) && !@git_lock.nil?
+
+    FileUtils.mkdir_p(LOCKFILE_DIR)
+
+    @git_lock = File.open(git_lock_filename, File::RDWR | File::CREAT, 0o644)
+    @git_lock.flock(File::LOCK_EX)                         # block waiting for lock
+    @git_lock.write("#{Process.pid} - #{Time.zone.now}\n") # for debugging
+    @git_lock.flush                                        # write current data
+    @git_lock.truncate(@git_lock.pos)                      # clean up remaining chars
+
+    true
+  end
+
+  def release_git_lock
+    return if !defined?(@git_lock) || @git_lock.nil?
```

Comment:
Do you need the defined check if you're going to check for nil anyway?

Reply:
Maybe it would be easier to make a `#git_lock` attr_accessor?

Reply:
> Maybe it would be easier to make a `#git_lock` attr_accessor?

@carbonin so I did respond to your earlier comments about this already:

https://github.com/ManageIQ/manageiq/pull/19074#discussion_r308969083
https://github.com/ManageIQ/manageiq/pull/19074#discussion_r308967973

However, I think I might have been misguided in my explanation or what you were suggestion.  I will take a look into this a bit later tonight, and I think that will probably help with my original issue.